### PR TITLE
Make test harness portable

### PR DESCRIPTION
**Pull request by KengoTODA**
Currently `AbstractIntegrationTest` in `test-harness` has several hard-coded value which depends on project structure of this repository, such as `"../findbugsTestCases"`, `"lib"` and `"/build/classes/main/"`. To make this library portable, it's better to remove them.

And one more problem, is that, we provide utilities as abstract class so it might be hard to use with another solution which uses the same way. Basically `Rule` is better choice for JUnit.

This pull request proposes:

* several refactorings
    * extract `CountMatcher`
    * extract `AnalysisRunner`
* create new `ExternalResource` named `SpotBugsRule`
* move `AbstractIntegrationTest` to `findbugs` subproject
    * this class shouldn't be a part of test-harness, because it contains hard-coded values
* use `SpotBugsRule` in `FindUnsatisfiedObligationTest`

### CODE REVIEW

File: test-harness/src/main/java/edu/umd/cs/findbugs/AnalysisRunner.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package edu.umd.cs.findbugs;
```
Comment (jsotuyod) [inline test-harness/src/main/java/edu/umd/cs/findbugs/AnalysisRunner.java]: we should probably move this to a more appropriate package... maybe `edu.umd.cs.findbugs.test`?

File: test-harness/src/main/java/edu/umd/cs/findbugs/CountMatcher.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package edu.umd.cs.findbugs;
```
Comment (jsotuyod) [inline test-harness/src/main/java/edu/umd/cs/findbugs/CountMatcher.java]: idem

File: test-harness/src/main/java/edu/umd/cs/findbugs/SpotBugsRule.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
+package edu.umd.cs.findbugs;
```
Comment (jsotuyod) [inline test-harness/src/main/java/edu/umd/cs/findbugs/SpotBugsRule.java]: idem

**Comment by KengoTODA** [comment]
@jsotuyod Renamed at c32aabc. This change needed to change visibility of several legacy classes, please check.

**Pull request merged**